### PR TITLE
Update install-rules.cmake

### PR DESCRIPTION
**Pull request by anders-wind**
The current install-rules double indent the include folder: as in `installation_prefix/include/libname/libname/header.hpp`. At least when running: `cmake --install build_folder_path --prefix=./installation_prefix`

Removing these three lines seems to be a better default behavior. When we tried to get our library on vcpkg, this was an issue which blocked us from merging.

### CODE REVIEW

**Comment by friendlyanon** [comment]
This is #43

Please see the conversation linked in there and reconsider.

> this was an issue which blocked us from merging.

The install include dir is a user-configurable cache variable and if you (or anyone packaging the project) wish to override this behavior, you may pass `-DCMAKE_INSTALL_INCLUDEDIR=include` to the configure command to achieve what you want in a cross-platform way.

**Comment by anders-wind** [comment]
Yea maybe - add some documentation on this explicitly - or maybe set the `CMAKE_INSTALL_INCLUDEDIR` to not just be package but `<package>-<version>` , that way its clear that it is two different concepts?

**Comment by anders-wind** [comment]
Edited the install-rules.cmake again to reflect my comment above - no idea if it would work as I don't know exactly how your template logic works.

**Review by friendlyanon — COMMENTED** [review]
I'm not sure how I feel about embedding the version. I guess anything is better than an unrelated system level dependency including the whole world anyway. The path is also customizable using built-in cache variables.

I'm curious, where did this cause an issue exactly? Can I take a look? Maybe that will give me a better insight hopefully.

Could you try your hand at maybe adding a section to BUILDING.md? Something that might be obvious to me might not be for someone else, so that could also help me better understand.

**Comment by anders-wind** [comment]
> I'm not sure how I feel about embedding the version. I guess anything is better than an unrelated system level dependency including the whole world anyway. The path is also customizable using built-in cache variables.
:D I think it seems like a really good idea to embed the version - You ofcourse decide what you want to do here, I'll try it out in our projects and see if we run into any issues.

> Could you try your hand at maybe adding a section to BUILDING.md? Something that might be obvious to me might not be for someone else, so that could also help me better understand.

> I'm curious, where did this cause an issue exactly? Can I take a look? Maybe that will give me a better insight hopefully.
I'll give it a go in adding some text in building - potentially also as a comment in the install_rules? 

So the final solution is exactly as you say: set CMAKE_INSTALL_INCLUDEDIR, but one of the maintainers of vcpkg had trouble using the library and started out by patching the project by deleting the line (the one I originally removed):
https://github.com/microsoft/vcpkg/pull/26740.

**Comment by friendlyanon** [comment]
Interesting, dg0yt did see the intended way to make things work the way you wanted, but curiously the suggestion to use relative paths did not make it into the port.

Another curiousity is this:

https://github.com/friendlyanon/cmake-init/blob/7a4c594eb82d679cbd80bb023a63c1ee030825c5/cmake-init/templates/common/cmake/install-rules.cmake#L44-L49

exists only so projects generated by cmake-init properly follow GNU conventions, but also provide a way for vcpkg specifically to control where the CMake package goes, as they always want the it to be in `share`.

**Comment by friendlyanon** [comment]
Hey, it's been 2 weeks. What's the status of this? I would like to make a release with the new warning flags and I would like to have the documentation changes from this PR in the release as well.

**Comment by anders-wind** [comment]
On vacation until tomorrow evening. I can add some documentation Wednesday :)

**Comment by anders-wind** [comment]
Ive added a comment about why we double indent, and a section on how to make a portfile.

File: cmake-init/templates/common/cmake/install-rules.cmake [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {% if not exe %}if(PROJECT_IS_TOP_LEVEL)
-  set(CMAKE_INSTALL_INCLUDEDIR include/{= name =} CACHE PATH "")
+  # CMAKE_INSTALL_INCLUDEDIR is nested in one more ${PROJECT_VERSION} directory to prevent inclusion of all other headers in the install prefix directory.
+  #   See https://github.com/friendlyanon/cmake-init/issues/43 for more info
```
Comment (friendlyanon) [inline cmake-init/templates/common/cmake/install-rules.cmake:3]: Besides not keeping the 80 column limit and the period at the end, I also don't feel okay about a link back to this project.

The idea of cmake-init is that it generates projects that - in a perfect world - everyone would write to begin with, so links back feel icky to me.

Something more generic about the purpose of avoiding indirect inclusion of other projects in a shared prefix would fit better.

File: cmake-init/templates/common/BUILDING.md [hunk 2]
```diff
@@ -79,6 +79,37 @@ target_link_libraries(
     {= name =}::{= name =}
 )
 ```
+
+### Creating a VCPKG port
```
Comment (friendlyanon) [inline cmake-init/templates/common/BUILDING.md:83]: "Officially" `vcpkg` is used like a regular word in its docs, so it shouldn't be all caps.

The additional directory in the include path also affects every other form of packaging, there is no need to make it vcpkg specific. Something like a "Note to packagers" section with cautioning them to review the usage of `CMAKE_INSTALL_INCLUDEDIR`. This should also be guarded with `{% if not exe %}`.

**Review by friendlyanon — CHANGES_REQUESTED** [review]

**Comment by friendlyanon** [comment]
I decided to implement this change, but I felt that the wording in this PR was excessive with the involvement of vcpkg.

I included you as a co-author, since you convinced me to add this change and opened a PR for it. Thanks!

**Pull request closed without merging**